scripts/src/mock_client.ts

```diff
@@ -1754,47 +1757,135 @@ async function checkValidTokenAfterLoginRightAfterLogout() {
 	console.log("OK.\n");
 }
 
+async function checkClientMigration() {
+	console.log("===================================== checkClientMigration =====================================");
+
+	const userId = generateId();
+	const deviceId = generateId();
+	const appClient = new SampleAppClient(SMPL_APP_CONFIG.jwtAddress);
+	const jwt = await appClient.getRegisterJWT(userId, deviceId);
+	const client = await MarketplaceClient.create({ jwt });
+
+	await axios.put(`${process.env.MARKETPLACE_BASE}/v2/applications/${client.appId}/blockchain_version`, {
+		blockchain_version: "2"
+	});
+	localCache.clear();
+
+	const keypair = Keypair.random();
+	await client.updateWallet(keypair.secret());
+
+	const selectedOfferV2 = await getOffer(client, "earn", "tutorial");
+
+	console.log(`requesting order for offer: ${ selectedOfferV2.id }: ${ selectedOfferV2.content.slice(0, 100) }`);
+	const openOrderV2 = await client.createOrder(selectedOfferV2.id);
+	console.log(`got order ${ openOrderV2.id }`);
+
+	const content = JSON.stringify({});
+
+	await client.submitOrder(openOrderV2.id, { content });
+	const orderV2 = await retry(() => client.getOrder(openOrderV2.id), order => order.status === "completed", "order did not turn completed");
+
+	console.log(`completion date: ${ orderV2.completion_date }`);
+	console.log(`got order after submit`, orderV2);
+
+	// change blockchain version to 3
+	const kin3BlockchainVersion: BlockchainVersion = "3";
+	await axios.put(`${process.env.MARKETPLACE_BASE}/v2/applications/${client.appId}/blockchain_version`, {
+		blockchain_version: kin3BlockchainVersion
+	});
+	client.requests.blockchainVersion = kin3BlockchainVersion;
+	localCache.clear();
```

Comment:
Calling this will clear the cache in your local testing process - it will not affect the marketplace-public server.
After calling `setBlockchainVersion` we either have to wait (as in `await delay(10)`) or we have to change marketplace server to not cache on non production environments